lib/src/soloud.dart

```diff
@@ -625,6 +658,166 @@ interface class SoLoud {
+  ///
+  /// [channels] enum to choose the number of channels.
+  ///
+  /// [pcmFormat] enum to choose from `f32le`, `s8`, `s16le` and `s32le`.
+  ///
+  /// [onBuffering] a callback that is called when starting to buffer
+  /// (isBuffering = true) and when the buffering is done (isBuffering = false).
+  /// It gives back the `handle` which triggered the event and the `time`
+  /// in seconds.
+  ///
+  /// Throws [SoLoudNotInitializedException] if the engine is not initialized.
+  AudioSource setBufferStream({
+    int maxBufferSize = 1024 * 1024 * 100, // 100 MB in bytes
+    double bufferingTimeNeeds = 2, // 2 seconds of data needed to un-pause
+    int sampleRate = 22050,
+    Channels channels = Channels.mono,
+    BufferPcmType pcmFormat = BufferPcmType.s16le,
+    void Function(bool isBuffering, int handle, double time)? onBuffering,
+  }) {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+
+    final ret = SoLoudController().soLoudFFI.setBufferStream(
+          maxBufferSize,
+          bufferingTimeNeeds,
+          sampleRate,
+          channels.count,
+          pcmFormat.value,
+          onBuffering,
+        );
+
+    final newSound = _addNewSound(ret.error, '', ret.soundHash.hash);
+    return newSound;
+  }
+
+  /// Add PCM audio data to the stream.
+  ///
+  /// This method can be called within an `Isolate` making possible
+  /// to create PCM data and send them to the buffer without frezing
+  /// the main thread.
+  /// When finishing to add data to the stream, call [setDataIsEnded].
+  ///
+  /// Throws [SoLoudNotInitializedException] if the engine is not initialized.
+  /// Throws [SoLoudPcmBufferFullOrStreamEndedCppException] if the buffer
+  /// is full or stream buffer has been set to be ended.
+  /// Thows [SoLoudOutOfMemoryException] if the buffer is out of OS memory or
+  /// the given `maxBufferSize` of the `setBufferStream` call is too small.
+  ///
+  /// See also:
+  ///
+  ///  * [addAudioDataStramF32] to add `Float32List` PCM audio data.
+  ///  * [addAudioDataStreamS8] to add `Int8List` PCM audio data.
+  ///  * [addAudioDataStreamS16] to add `Int16List` PCM audio data.
+  ///  * [addAudioDataStreamS32] to add `Int32List` PCM audio data.
+  void addAudioDataStreamU8(
+    AudioSource source,
+    Uint8List audioChunk,
+  ) {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+
+    final e = SoLoudController().soLoudFFI.addAudioDataStream(
+          source.soundHash.hash,
+          audioChunk,
+        );
+
+    if (e != PlayerErrors.noError) {
+      _logPlayerError(e, from: 'addAudioDataStream() result');
+      throw SoLoudCppException.fromPlayerError(e);
+    }
+  }
+
+  /// Helper method to add `Float32List` PCM audio data to the stream.
+  ///
+  /// This method is a wrapper around [addAudioDataStreamU8].
+  void addAudioDataStramF32(AudioSource source, Float32List audioChunk) {
```

Comment:
I wonder if it's worth it to have all these help methods. It pollutes the API with a bunch of members. Maybe we should just teach people how to "cast" their typed data into raw `UInt8List`? And then you only have one method, `addAudioDataStream()` (without the `U8`).

Reply:
And by "teach people how to...", I mean adding sample code into `addAudioDataStream()`'s API doc.

Reply:
Good point. I was concerned by this because other typed data are not foreseen ie `Int32x4List`. But anyway, they can be casted to Uint8List, so it's good to have only one method.
Done.